contracts/Nexus.sol

```diff
@@ -47,11 +46,18 @@ contract Nexus is INexus, EIP712, BaseAccount, ExecutionHelper, ModuleManager, U
     ///         - An EIP-712 hash: keccak256("\x19\x01" || someDomainSeparator || hashStruct(someStruct))
     bytes32 private constant _MESSAGE_TYPEHASH = keccak256("BiconomyNexusMessage(bytes32 hash)");
 
+    address private immutable _SELF;
```

Comment:
what is the purpose of this variable ? it's used nowhere

Reply:
it could be used to differentiate/ making sure calls coming from proxy vs implementation
in receive function could check
if (address(this) == SELF) revert

Reply:
ok 👍! in this case let's update the code to use it and let's add a natspec to describe this state var

Reply:
not sure if we will use it right away cause someone reported receive function with anything else added starts failing for calls made with 2300 stipend.